File: test_db2.mjs
import { createClient } from '@supabase/supabase-js';
import dotenv from 'dotenv';
dotenv.config();

const supabaseUrl = process.env.SUPABASE_URL || process.env.NEXT_PUBLIC_SUPABASE_URL;
const supabaseKey = process.env.SUPABASE_SERVICE_ROLE_KEY || process.env.SUPABASE_KEY;
const supabase = createClient(supabaseUrl, supabaseKey);

async function main() {
  const { data: tps } = await supabase.from('team_players').select('team_id, matchday, is_starter').eq('is_starter', true);
  
  const countsByTeamAndMd = {};
  tps.forEach(tp => {
    if (!countsByTeamAndMd[tp.team_id]) countsByTeamAndMd[tp.team_id] = {};
    countsByTeamAndMd[tp.team_id][tp.matchday] = (countsByTeamAndMd[tp.team_id][tp.matchday] || 0) + 1;
  });
  
  for (const [tid, mds] of Object.entries(countsByTeamAndMd)) {
    for (const [md, count] of Object.entries(mds)) {
      if (count !== 11) {
        console.log(`Team ${tid} Matchday ${md} has ${count} starters!`);
      }
    }
  }
}
main();

File: test_db2.py
import os
from supabase import create_client
from dotenv import load_dotenv

load_dotenv('frontend-web/.env.local')
load_dotenv('frontend-web/.env')

url = os.environ.get("SUPABASE_URL") or os.environ.get("NEXT_PUBLIC_SUPABASE_URL")
key = os.environ.get("SUPABASE_SERVICE_ROLE_KEY")
client = create_client(url, key)

res = client.table('player_scores').select('id, match_id, player_id, calidad_parada, relevo_block_2_pts, saves, minutes_played').eq('position', 'POR').execute()
for r in res.data:
    if r['calidad_parada'] and r['calidad_parada'] > 0:
        val_per_min = r['calidad_parada'] / r['minutes_played'] if r['minutes_played'] > 0 else 0
        if val_per_min > 0.006:
            print(f"ID: {r['id']}, Calidad: {r['calidad_parada']}, Mins: {r['minutes_played']}, Val/m: {val_per_min:.4f}, Block 2 pts: {r['relevo_block_2_pts']}")
